helion/language/memory_ops.py

```diff
@@ -463,3 +468,56 @@ def _(state: CodegenState) -> ast.AST:
         mask=indices.mask_expr,
         sem=sem,
     )
+
+
+@_decorators.device_func_replacement(torch.stack)
```

Comment:
`torch.stack` lowering with torch.compile generates `ConcatKernel` which is actually a NopKernel that needs to be fused with the next op. In Helion, my understanding is that each statement is an op by itself and the output needs to be materialized. So we need this function replacement to get the materialized output tensor of torch.stack.